Feature - Add divider collection

DavidLambauer opened the pull request:
Add a new divider collection to the application UI category.

DavidLambauer commented:
Starting with the two components. Want to add at least left/right-oriented text dividers, too.

markmead commented on public/components/application-ui-dividers/1.html:
```diff
@@ -0,0 +1,5 @@
+<div class="flex flex-row items-center max-w-7xl mx-auto">
```
Safe to remove these classes `flex-row max-w-7xl mx-auto`

markmead commented on public/components/application-ui-dividers/1.html:
```diff
@@ -0,0 +1,5 @@
+<div class="flex flex-row items-center max-w-7xl mx-auto">
+    <div class="w-full flex border-black border-t ml-12"></div>
+    <span class="px-6">more</span>
+    <span class="w-full flex border-black border-t mr-12"></span>
```
These could all be `<span>`

markmead replied:
This could be `flex-1 h-px bg-black`.

markmead commented on public/components/application-ui-dividers/1.html:
```diff
@@ -0,0 +1,5 @@
+<div class="flex flex-row items-center max-w-7xl mx-auto">
+    <div class="w-full flex border-black border-t ml-12"></div>
```
This could be `flex-1 h-px bg-black`.

markmead commented on public/components/application-ui-dividers/2.html:
```diff
@@ -0,0 +1 @@
+<div class="h-px border-t-0 bg-transparent bg-gradient-to-r from-transparent via-neutral-500 to-transparent opacity-75"></div>
```
No need for the border classes.

markmead commented on src/data/components/application-ui-dividers.mdx:
```diff
@@ -0,0 +1,22 @@
+emoji: 🗡️
+category: application-ui
+tag: new
+container: py-20 max-w-3xl mx-auto items-center
```
Maybe `p-6 grid place-content-center`.

markmead reviewed (changes requested):
Looks good! Just a few small changes.

markmead commented on public/components/application-ui-dividers/1.html:
```diff
@@ -0,0 +1,5 @@
+<div class="flex flex-row items-center max-w-7xl mx-auto">
+    <div class="w-full flex border-black border-t ml-12"></div>
+    <span class="px-6">more</span>
```
Be good to put a `shrink-0` on this as well.

DavidLambauer commented:
Thx for the comments. Will continue working on it soon! 🥳

DavidLambauer commented:
HI @markmead, 

Thank you for the feedback. I'm actually a backend dev and still learning proper FE development. Glad you take the time to review it. 

I guess I addressed your feedback and added two more dividers for left/right alignment. Not sure about the `place-content-center` thing, though. Doesn't seem to have the correct effect.

markmead commented on public/components/application-ui-dividers/1.html:
```diff
@@ -0,0 +1,5 @@
+<span class="flex items-center">
+    <span class="flex-1 bg-black h-px ml-12"></span>
+    <span class="px-6 shrink-0">more</span>
+    <span class="flex-1 bg-black h-px mr-12"></span>
```
Think it's OK to remove the margin classes.

markmead commented on public/components/application-ui-dividers/3.html:
```diff
@@ -0,0 +1,4 @@
+<span class="flex items-center">
+    <span class="px-4">Align Left</span>
```
Should use `pr` instead of `px` and let's use `pr-6` so it is consistent spacing with the first one.

markmead commented on public/components/application-ui-dividers/3.html:
```diff
@@ -0,0 +1,4 @@
+<span class="flex items-center">
+    <span class="px-4">Align Left</span>
+    <span class="flex-1 bg-black h-px mr-12"></span>
```
Can remove the `mr-12`.

markmead commented on public/components/application-ui-dividers/4.html:
```diff
@@ -0,0 +1,4 @@
+<span class="flex items-center">
+    <span class="flex-1 bg-black h-px ml-12"></span>
+    <span class="px-4">Align Right</span>
+</span>
```
Same comments as above just applied to this component.

markmead commented on src/data/components/application-ui-dividers.mdx:
```diff
@@ -0,0 +1,26 @@
+  2:
+    title: Blurry Divider
+  3:
+    title: Algin Left
```
Algin > Align

markmead reviewed (changes requested):
Good stuff! Small changes and then we should be good to go, unless you want to keep add some more?

markmead commented:
Any updates for this? @DavidLambauer

DavidLambauer commented:
Will work on it tonight. No more dividers planned

DavidLambauer commented:
@markmead done 😄 Thanks for the reminder.
